code/modules/mob/living/simple_animal/hostile/mining_mobs/elites/elite.dm

```diff
@@ -0,0 +1,354 @@
+						addtimer(CALLBACK(src, .proc/spawn_elite, elitemind), 100)
+						return
+					else
+						visible_message("<span class='boldwarning'>The stirring stops, and nothing emerges.  Perhaps try again later.</span>")
+						activity = TUMOR_INACTIVE
+						activator = null
+						return
+				else
+					addtimer(CALLBACK(src, .proc/spawn_elite), 30)
+				return
+			if(TUMOR_PASSIVE)
+				activity = TUMOR_ACTIVE
+				visible_message("<span class='boldwarning'>The tumor convulses as your arm enters its radius.  Your instincts tell you to step back.</span>")
+				activator = user
+				INVOKE_ASYNC(src, .proc/arena_trap)
+				if(boosted)
+					SEND_SOUND(mychild, sound('sound/effects/magic.ogg'))
+					to_chat(mychild, "<b>Someone has activated your tumor.  You will be returned to fight shortly, get ready!</b>")
+				addtimer(CALLBACK(src, .proc/return_elite), 30)
+				return
+
+				
+obj/structure/elite_tumor/proc/spawn_elite(var/mob/dead/observer/elitemind)
+	var/selectedspawn = pick(potentialspawns)
+	mychild = new selectedspawn(loc)
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	if(boosted)
+		mychild.key = elitemind.key
+		mychild.sentience_act()
+	INVOKE_ASYNC(src, .proc/arena_trap)
+	return
+
+obj/structure/elite_tumor/proc/return_elite()
+	mychild.forceMove(loc)
+	visible_message("<span class='boldwarning'>[mychild] emerges from the tumor!</span>")
+	playsound(loc,'sound/effects/phasein.ogg', 200, 0, 50, TRUE, TRUE)
+	mychild.revive(full_heal = TRUE, admin_revive = TRUE)
+	if(boosted)
+		mychild.maxHealth = mychild.maxHealth * 2
+		mychild.health = mychild.maxHealth
+	return
+		
+/obj/structure/elite_tumor/Initialize()
+	. = ..()
+	AddComponent(/datum/component/gps, "Menacing Signal")
+	START_PROCESSING(SSobj, src)
+	
+/obj/structure/elite_tumor/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	mychild = null
+	return ..()
+	
+/obj/structure/elite_tumor/process()
+	if(isturf(loc))
+		for(var/mob/living/simple_animal/hostile/asteroid/elite/elitehere in loc)
+			if(elitehere == mychild && activity == TUMOR_PASSIVE)
+				mychild.adjustHealth(-mychild.maxHealth*0.05)
+				var/obj/effect/temp_visual/heal/H = new /obj/effect/temp_visual/heal(get_turf(mychild))
+				H.color = "#FF0000"
+				mychild.update_health_hud()
+		
+/obj/structure/elite_tumor/attackby(obj/item/W, mob/user, params)
```

Comment:
better to make a cast var like this end with whatever is last used, in this case `item` -> `I`.